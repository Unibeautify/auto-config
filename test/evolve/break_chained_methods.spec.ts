import { testGenetic } from "../testGenetic";
import prettier from "@unibeautify/beautifier-prettier";
import jsBeautify from "@unibeautify/beautifier-js-beautify";

describe("break_chained_methods", () => {
  testGenetic("break_chained_methods=true", {
    beautifiers: [jsBeautify],
    language: "JavaScript",
    originalText: `foo.bar().baz();`,
    desiredText: `foo.bar()\n  .baz();`,
  });

  testGenetic(
    "break_chained_methods1",
    {
      beautifiers: [prettier, jsBeautify],
      language: "JavaScript",
      originalText: `this.$("#fileName").val().addClass("disabled")\n  .prop("disabled", true)`,
      desiredText: `this.$("#fileName")\n  .val()\n  .addClass("disabled")\n  .prop("disabled", true)\n`,
    },
    true
  );

  testGenetic(
    "break_chained_methods2",
    {
      beautifiers: [prettier, jsBeautify],
      language: "JavaScript",
      originalText: `this.$("#fileName").val().addClass("disabled")\n  .prop("disabled", true)`,
      desiredText: `this.$("#fileName").val().addClass("disabled")\n  .prop("disabled", true)`,
    },
    true
  );
});
