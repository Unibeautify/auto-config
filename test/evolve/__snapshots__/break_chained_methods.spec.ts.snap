// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`break_chained_methods break_chained_methods=true: beautifiedText 1`] = `
foo.bar()
  .baz();
`;

exports[`break_chained_methods break_chained_methods=true: entity 1`] = `
Object {
  "options": Object {
    "beautifiers": Array [
      "JS-Beautify",
    ],
    "indent_size": 2,
    "wrap_line_length": 8,
  },
}
`;

exports[`break_chained_methods break_chained_methods=true: fitness 1`] = `1`;
